<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2024-03-19" />

<title>HumanAneurysm</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HumanAneurysm</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">HumanAneurysm</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2024-03-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>HumanAneurysm/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240319code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240319code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240319)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataMarch2024/
</td>
<td style="text-align:left;">
data/AneurysmDataMarch2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataApril2024/
</td>
<td style="text-align:left;">
data/AneurysmDataApril2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataJuly2024/
</td>
<td style="text-align:left;">
data/AneurysmDataJuly2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataAugust2024/
</td>
<td style="text-align:left;">
data/AneurysmDataAugust2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/Aneuryms_allmerged_seurat.rds
</td>
<td style="text-align:left;">
data/Aneuryms_allmerged_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenes
</td>
<td style="text-align:left;">
analysis/markerGenes
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenesintbatch
</td>
<td style="text-align:left;">
analysis/markerGenesintbatch
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenesintpat
</td>
<td style="text-align:left;">
analysis/markerGenesintpat
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinHumanAneurysmtreed91538dcf89555aae858fc6bfcc58c965538e570targetblankd91538da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/HumanAneurysm/tree/d91538dcf89555aae858fc6bfcc58c965538e570" target="_blank">d91538d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinHumanAneurysmtreed91538dcf89555aae858fc6bfcc58c965538e570targetblankd91538da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/HumanAneurysm/tree/d91538dcf89555aae858fc6bfcc58c965538e570" target="_blank">d91538d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/AneurysmDataApril2024/
    Ignored:    data/AneurysmDataAugust2024/
    Ignored:    data/AneurysmDataJuly2024/
    Ignored:    data/AneurysmDataMarch2024/

Untracked files:
    Untracked:  analysis/markerGenesintbatch

Unstaged changes:
    Modified:   analysis/HumanAneurysm.Rmd
    Modified:   analysis/QCindividual.Rmd
    Modified:   analysis/markerGenes

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/HumanAneurysm.Rmd</code>) and
HTML (<code>docs/HumanAneurysm.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/analysis/HumanAneurysm.Rmd" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
<td>
update analysise
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/HumanAneurysm/d91538dcf89555aae858fc6bfcc58c965538e570/docs/HumanAneurysm.html" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
<td>
update analysise
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/HumanAneurysm/d6526d4ac2dc949b6fc2e4608d44487e8541529c/docs/HumanAneurysm.html" target="_blank">d6526d4</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-04-29
</td>
<td>
update gitignore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/4ede22322d35e80ff492ca3fdef8e772876854c7/analysis/HumanAneurysm.Rmd" target="_blank">4ede223</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-04-29
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<p>##############################start
pre-processing##############################</p>
</div>
<div id="load-files-and-merge" class="section level2">
<h2>load files and merge</h2>
<pre class="r"><code>### load and merge all 
basedir &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataMarch2024/&quot;
fileNamList &lt;- list.files(path = basedir)

for(i in 1:length(fileNamList)){
  seuratS &lt;- readRDS(paste0(basedir, fileNamList[i]))
  if(exists(&quot;seuratM&quot;)){
    seuratM &lt;- merge(x = seuratM, y = seuratS)
  }else{
    seuratM &lt;- seuratS
  }
}

remove(seuratS)
table(seuratM$dataset)</code></pre>
<pre><code>
344261_07-7_20240219_Hu_nucseq_control_01     344261_08-8_20240219_Hu_nucseq_UIA_04 
                                      598                                       398 
    344261_09-9_20240219_Hu_nucseq_UIA_07 
                                     1132 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
     
2128 </code></pre>
<pre class="r"><code>### load and merge all 
basedir &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataApril2024/&quot;
fileNamList &lt;- list.files(path = basedir)

for(i in 1:length(fileNamList)){
  seuratS &lt;- readRDS(paste0(basedir, fileNamList[i]))
  if(exists(&quot;seuratM&quot;)){
    seuratM &lt;- merge(x = seuratM, y = seuratS)
  }else{
    seuratM &lt;- seuratS
  }
}

remove(seuratS)
table(seuratM$dataset)</code></pre>
<pre><code>
344261_07-7_20240219_Hu_nucseq_control_01     344261_08-8_20240219_Hu_nucseq_UIA_04 
                                      598                                       398 
    344261_09-9_20240219_Hu_nucseq_UIA_07 348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1 
                                     1132                                      1081 
348751_3-8_20240326_Hu_nucseq_tArm_ctrl_2     348751_4-9_20240326_Hu_nucseq_UIA_001 
                                      798                                       351 
   348751_5-10_20240326_Hu_nucseq_UIA_003    348751_6-11_20240326_Hu_nucseq_UIA_005 
                                     5107                                     13226 
   348751_7-12_20240326_Hu_nucseq_UIA_012    348751_8-13_20240326_Hu_nucseq_UIA_002 
                                     1083                                     23093 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
46867 </code></pre>
<pre class="r"><code>### load and merge all 
basedir &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataJuly2024/&quot;
fileNamList &lt;- list.files(path = basedir)

for(i in 1:length(fileNamList)){
  seuratS &lt;- readRDS(paste0(basedir, fileNamList[i]))
  if(exists(&quot;seuratM&quot;)){
    seuratM &lt;- merge(x = seuratM, y = seuratS)
  }else{
    seuratM &lt;- seuratS
  }
}

### load and merge all 
basedir &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/AneurysmDataAugust2024/&quot;
fileNamList &lt;- list.files(path = basedir)

for(i in 1:length(fileNamList)){
  seuratS &lt;- readRDS(paste0(basedir, fileNamList[i]))
  if(exists(&quot;seuratM&quot;)){
    seuratM &lt;- merge(x = seuratM, y = seuratS)
  }else{
    seuratM &lt;- seuratS
  }
}

remove(seuratS)
table(seuratM$dataset)</code></pre>
<pre><code>
      344261_07-7_20240219_Hu_nucseq_control_01           344261_08-8_20240219_Hu_nucseq_UIA_04 
                                            598                                             398 
          344261_09-9_20240219_Hu_nucseq_UIA_07       348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1 
                                           1132                                            1081 
      348751_3-8_20240326_Hu_nucseq_tArm_ctrl_2           348751_4-9_20240326_Hu_nucseq_UIA_001 
                                            798                                             351 
         348751_5-10_20240326_Hu_nucseq_UIA_003          348751_6-11_20240326_Hu_nucseq_UIA_005 
                                           5107                                           13226 
         348751_7-12_20240326_Hu_nucseq_UIA_012          348751_8-13_20240326_Hu_nucseq_UIA_002 
                                           1083                                           23093 
353921_25-25_20240529_Hu_nucseq_Cadaver_ctrl_01          353921_26-26_20240529_Hu_nucseq_STA_LG 
                                           2270                                            1605 
    353921_27-27_20240522_Hu_nucseq_Ocp_ctrl_03     359861_19-11_20240730_Hu_nucseq_Cad39_Aneur 
                                            487                                            2116 
    359861_20-12_20240730_Hu_nucseq_Cad23_Aneur 
                                           1454 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
54799 </code></pre>
</div>
<div id="add-metadata" class="section level2">
<h2>add metadata</h2>
<pre class="r"><code>### add any type of metadata
### patient
ctrl01 &lt;- c(&quot;344261_07-7_20240219_Hu_nucseq_control_01&quot;)
UIA04 &lt;- c(&quot;344261_08-8_20240219_Hu_nucseq_UIA_04&quot;)
UIA07 &lt;- c(&quot;344261_09-9_20240219_Hu_nucseq_UIA_07&quot;)
tArmctrl1 &lt;- c(&quot;348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1&quot;)
tArmctrl2 &lt;- c(&quot;348751_3-8_20240326_Hu_nucseq_tArm_ctrl_2&quot;)
UIA001 &lt;- c(&quot;348751_4-9_20240326_Hu_nucseq_UIA_001&quot;)
UIA3 &lt;- c(&quot;348751_5-10_20240326_Hu_nucseq_UIA_003&quot;)
UIA5 &lt;- c(&quot;348751_6-11_20240326_Hu_nucseq_UIA_005&quot;)
UIA012 &lt;- c(&quot;348751_7-12_20240326_Hu_nucseq_UIA_012&quot;)
UIA002 &lt;- c(&quot;348751_8-13_20240326_Hu_nucseq_UIA_002&quot;)
Cadctrl01 &lt;- c(&quot;353921_25-25_20240529_Hu_nucseq_Cadaver_ctrl_01&quot;)
STALG &lt;- c(&quot;353921_26-26_20240529_Hu_nucseq_STA_LG&quot;)
Ocpctrl03 &lt;- c(&quot;353921_27-27_20240522_Hu_nucseq_Ocp_ctrl_03&quot;)
Cad23 &lt;- c(&quot;359861_20-12_20240730_Hu_nucseq_Cad23_Aneur&quot;)
Cad39 &lt;- c(&quot;359861_19-11_20240730_Hu_nucseq_Cad39_Aneur&quot;)

seuratM$patient &lt;- &quot;pat_nr&quot;
seuratM$patient[which(seuratM$dataset %in% ctrl01)] &lt;- &quot;ctrl01&quot;
seuratM$patient[which(seuratM$dataset %in% UIA04)] &lt;- &quot;UIA04&quot;
seuratM$patient[which(seuratM$dataset %in% UIA07)] &lt;- &quot;UIA07&quot;
seuratM$patient[which(seuratM$dataset %in% tArmctrl1)] &lt;- &quot;tArmctrl1&quot;
seuratM$patient[which(seuratM$dataset %in% tArmctrl2)] &lt;- &quot;tArmctrl2&quot;
seuratM$patient[which(seuratM$dataset %in% UIA001)] &lt;- &quot;UIA001&quot;
seuratM$patient[which(seuratM$dataset %in% UIA3)] &lt;- &quot;UIA3&quot;
seuratM$patient[which(seuratM$dataset %in% UIA5)] &lt;- &quot;UIA5&quot;
seuratM$patient[which(seuratM$dataset %in% UIA012)] &lt;- &quot;UIA012&quot;
seuratM$patient[which(seuratM$dataset %in% UIA002)] &lt;- &quot;UIA002&quot;
seuratM$patient[which(seuratM$dataset %in% Cadctrl01)] &lt;- &quot;Cadctrl01&quot;
seuratM$patient[which(seuratM$dataset %in% STALG)] &lt;- &quot;STALG&quot;
seuratM$patient[which(seuratM$dataset %in% Ocpctrl03)] &lt;- &quot;Ocpctrl03&quot;
seuratM$patient[which(seuratM$dataset %in% Cad23)] &lt;- &quot;Cad23&quot;
seuratM$patient[which(seuratM$dataset %in% Cad39)] &lt;- &quot;Cad39&quot;
table(seuratM$patient)</code></pre>
<pre><code>
    Cad23     Cad39 Cadctrl01    ctrl01 Ocpctrl03     STALG tArmctrl1 tArmctrl2    UIA001    UIA002 
     1454      2116      2270       598       487      1605      1081       798       351     23093 
   UIA012     UIA04     UIA07      UIA3      UIA5 
     1083       398      1132      5107     13226 </code></pre>
<pre class="r"><code>### batch
batchMarch24 &lt;- c(&quot;344261_07-7_20240219_Hu_nucseq_control_01&quot;,&quot;344261_08-8_20240219_Hu_nucseq_UIA_04&quot;,&quot;344261_09-9_20240219_Hu_nucseq_UIA_07&quot;)
batchApril24 &lt;- c(&quot;348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1&quot;,&quot;348751_3-8_20240326_Hu_nucseq_tArm_ctrl_2&quot;,&quot;348751_4-9_20240326_Hu_nucseq_UIA_001&quot;, &quot;348751_5-10_20240326_Hu_nucseq_UIA_003&quot;, &quot;348751_6-11_20240326_Hu_nucseq_UIA_005&quot;, &quot;348751_7-12_20240326_Hu_nucseq_UIA_012&quot;,&quot;348751_8-13_20240326_Hu_nucseq_UIA_002&quot;)
batchJuly24 &lt;- c(&quot;353921_25-25_20240529_Hu_nucseq_Cadaver_ctrl_01&quot;,&quot;353921_26-26_20240529_Hu_nucseq_STA_LG&quot;,&quot;353921_27-27_20240522_Hu_nucseq_Ocp_ctrl_03&quot;)
batchAugust24 &lt;- c(&quot;359861_20-12_20240730_Hu_nucseq_Cad23_Aneur&quot;,&quot;359861_19-11_20240730_Hu_nucseq_Cad39_Aneur&quot;)

seuratM$batch &lt;- &quot;batch&quot;
seuratM$batch[which(seuratM$dataset %in% batchMarch24)] &lt;- &quot;batchMarch24&quot;
seuratM$batch[which(seuratM$dataset %in% batchApril24)] &lt;- &quot;batchApril24&quot;
seuratM$batch[which(seuratM$dataset %in% batchJuly24)] &lt;- &quot;batchJuly24&quot;
seuratM$batch[which(seuratM$dataset %in% batchAugust24)] &lt;- &quot;batchAugust24&quot;
table(seuratM$batch)</code></pre>
<pre><code>
 batchApril24 batchAugust24   batchJuly24  batchMarch24 
        44739          3570          4362          2128 </code></pre>
<pre class="r"><code>### source
surgery &lt;- c(&quot;344261_07-7_20240219_Hu_nucseq_control_01&quot;,&quot;344261_08-8_20240219_Hu_nucseq_UIA_04&quot;,&quot;344261_09-9_20240219_Hu_nucseq_UIA_07&quot;,&quot;348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1&quot;,&quot;348751_3-8_20240326_Hu_nucseq_tArm_ctrl_2&quot;,&quot;348751_4-9_20240326_Hu_nucseq_UIA_001&quot;, &quot;348751_5-10_20240326_Hu_nucseq_UIA_003&quot;, &quot;348751_6-11_20240326_Hu_nucseq_UIA_005&quot;, &quot;348751_7-12_20240326_Hu_nucseq_UIA_012&quot;,&quot;348751_8-13_20240326_Hu_nucseq_UIA_002&quot;,&quot;353921_26-26_20240529_Hu_nucseq_STA_LG&quot;,&quot;353921_27-27_20240522_Hu_nucseq_Ocp_ctrl_03&quot;)
autopsy &lt;- c(&quot;353921_25-25_20240529_Hu_nucseq_Cadaver_ctrl_01&quot;, &quot;359861_20-12_20240730_Hu_nucseq_Cad23_Aneur&quot;, &quot;359861_19-11_20240730_Hu_nucseq_Cad39_Aneur&quot;)

seuratM$source &lt;- &quot;source&quot;
seuratM$source[which(seuratM$dataset %in% surgery)] &lt;- &quot;surgery&quot;
seuratM$source[which(seuratM$dataset %in% autopsy)] &lt;- &quot;autopsy&quot;
table(seuratM$source)</code></pre>
<pre><code>
autopsy surgery 
   5840   48959 </code></pre>
<p>##set color vectors</p>
<pre class="r"><code>col &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;,&quot;#727077&quot;,&quot;#355C7D&quot;,&quot;#779d8d&quot;,&quot;#dfc27d&quot;,&quot;#f4a582&quot;,&quot;#B45B5C&quot;)
names(col) &lt;- c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;)

#get colors
#brewer.pal(n= 10, name = &quot;Spectral&quot;)
#&quot;#9E0142&quot; &quot;#D53E4F&quot; &quot;#F46D43&quot; &quot;#FDAE61&quot; &quot;#FEE08B&quot; &quot;#E6F598&quot; &quot;#ABDDA4&quot; &quot;#66C2A5&quot; &quot;#3288BD&quot; &quot;#5E4FA2&quot;
#brewer.pal(n= 10, name = &quot;Accent&quot;)
#&quot;#7FC97F&quot; &quot;#BEAED4&quot; &quot;#FDC086&quot; &quot;#FFFF99&quot; &quot;#386CB0&quot; &quot;#F0027F&quot; &quot;#BF5B17&quot; &quot;#666666&quot;

colpat &lt;- c(&quot;#9E0142&quot;, &quot;#D53E4F&quot;, &quot;#F46D43&quot;, &quot;#FDAE61&quot;, &quot;#FEE08B&quot;, &quot;#E6F598&quot;, &quot;#ABDDA4&quot;, &quot;#66C2A5&quot;, &quot;#3288BD&quot;, &quot;#5E4FA2&quot; ,&quot;#BEAED4&quot; , &quot;#FDC086&quot;, &quot;#727077&quot;,&quot;#B45B5C&quot;,&quot;#202547&quot;)
names(colpat) &lt;- unique(seuratM$patient)

colbatch &lt;- c(&quot;#779d8d&quot;,&quot;#dfc27d&quot;,&quot;#f4a582&quot;,&quot;#D53E4F&quot;)
names(colbatch) &lt;- unique(seuratM$batch)

colsource &lt;- c(&quot;#779d8d&quot;,&quot;#f4a582&quot;)
names(colsource) &lt;- unique(seuratM$source)</code></pre>
</div>
<div id="qc-merged" class="section level2">
<h2>QC merged</h2>
<pre class="r"><code>### QC merged
# Extract meta.data from the Seurat object
meta.data &lt;- seuratM@meta.data
# Create the density plot
ptotalpat &lt;- ggplot(data = meta.data, aes(x = total, color = patient, fill = patient)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colpat) +
  scale_color_manual(values = colpat) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)

pdetectedpat &lt;- ggplot(data = meta.data, aes(x = detected, color = patient, fill = patient)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colpat) +
  scale_color_manual(values = colpat) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)

ptotalbatch &lt;- ggplot(data = meta.data, aes(x = total, color = batch, fill = batch)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colbatch) +
  scale_color_manual(values = colbatch) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)

pdetectedbatch &lt;- ggplot(data = meta.data, aes(x = detected, color = batch, fill = batch)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colbatch) +
  scale_color_manual(values = colbatch) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)

ptotalsource &lt;- ggplot(data = meta.data, aes(x = total, color = source, fill = source)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colsource) +
  scale_color_manual(values = colsource) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)

pdetectedsource &lt;- ggplot(data = meta.data, aes(x = detected, color = source, fill = source)) +
  geom_density(alpha = 0.2) +
  scale_fill_manual(values = colsource) +
  scale_color_manual(values = colsource) +
  theme_classic() +
  scale_x_log10() +
  ylab(&quot;density&quot;) +
  geom_vline(xintercept = 100)
# Return the plots as a list
list(ptotalpat, pdetectedpat, ptotalbatch ,pdetectedbatch, ptotalsource, pdetectedsource)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-QC-merged-1">
Past versions of "QC merged-1.png"
</button>
</p>
<div id="fig-no-spaces-QC-merged-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/QC merged-1.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-QC-merged-2">
Past versions of "QC merged-2.png"
</button>
</p>
<div id="fig-no-spaces-QC-merged-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/QC merged-2.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-QC-merged-3">
Past versions of "QC merged-3.png"
</button>
</p>
<div id="fig-no-spaces-QC-merged-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/QC merged-3.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-QC-merged-4">
Past versions of "QC merged-4.png"
</button>
</p>
<div id="fig-no-spaces-QC-merged-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/QC merged-4.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/QC%20merged-6.png" width="672" style="display: block; margin: auto;" />
## filter and rerun seurat</p>
<pre class="r"><code>##filter for Cadaver_ctr_01, UIA_07, UIA_04, Ocp_ctrl_03, control_01, tArm_ctr_1
seuratMfil &lt;- subset(seuratM, patient %in% c(&quot;Cadctrl01&quot;, &quot;UIA07&quot;, &quot;UIA04&quot;, &quot;Ocpctrl03&quot;, &quot;ctrl01&quot;, &quot;tArmctrl1&quot;, &quot;Cad23&quot;, &quot;Cad39&quot;))
table(seuratMfil$dataset)

seuratM &lt;- seuratMfil
remove(seuratMfil)
table(seuratM$dataset)

#rerun seurat
seuratM &lt;- NormalizeData (object = seuratM)
seuratM &lt;- FindVariableFeatures(object = seuratM)
seuratM &lt;- ScaleData(object = seuratM, verbose = TRUE)
seuratM &lt;- RunPCA(object=seuratM, npcs = 30, verbose = FALSE)
seuratM &lt;- RunTSNE(object=seuratM, reduction=&quot;pca&quot;, dims = 1:20)
seuratM &lt;- RunUMAP(object=seuratM, reduction=&quot;pca&quot;, dims = 1:20)
seuratM &lt;- FindNeighbors(object = seuratM, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratM &lt;- FindClusters(object = seuratM, resolution = res[i], random.seed = 1234)
}</code></pre>
</div>
<div id="add-metadata-2" class="section level2">
<h2>add metadata-2</h2>
<pre class="r"><code>### diseasecond
ICctrl &lt;- c(&quot;Cadctrl01&quot;, &quot;Cad23&quot;)
ECctrl &lt;- c(&quot;Ocpctrl03&quot;, &quot;ctrl01&quot;, &quot;tArmctrl1&quot;)
aneurysm &lt;- c(&quot;Cad39&quot;,&quot;UIA07&quot;, &quot;UIA04&quot;)

seuratM$diseasecond &lt;- &quot;diseasecond&quot;
seuratM$diseasecond[which(seuratM$patient %in% ICctrl)] &lt;- &quot;ICctrl&quot;
seuratM$diseasecond[which(seuratM$patient %in% ECctrl)] &lt;- &quot;ECctrl&quot;
seuratM$diseasecond[which(seuratM$patient %in% aneurysm)] &lt;- &quot;aneurysm&quot;
table(seuratM$diseasecond)

### order patients
seuratM$patient &lt;- factor(seuratM$patient, levels = c(&quot;Ocpctrl03&quot;, &quot;ctrl01&quot;, &quot;tArmctrl1&quot;, &quot;Cadctrl01&quot;, &quot;Cad23&quot;, &quot;Cad39&quot;,&quot;UIA07&quot;, &quot;UIA04&quot;))
table(seuratM$patient)</code></pre>
<pre class="r"><code>### save seurat object
saveRDS(seuratM, file=&quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/Aneuryms_allmerged_seurat.rds&quot;)</code></pre>
<p>##############################end
pre-processing##############################</p>
</div>
<div id="load-file" class="section level2">
<h2>load file</h2>
<pre class="r"><code>##load merged file 
fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/data/Aneuryms_allmerged_seurat.rds&quot;
seuratM &lt;- readRDS(fileNam)

colpat2 &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;,&quot;#727077&quot;,&quot;#779d8d&quot;,&quot;#dfc27d&quot;,&quot;#f4a582&quot;,&quot;#B45B5C&quot;,&quot;#ABDDA4&quot;)
names(colpat2) &lt;- unique(seuratM$patient)

coldiseasecond &lt;- c(&quot;#ABDDA4&quot;,&quot;#B45B5C&quot;,&quot;#779d8d&quot;)
names(coldiseasecond) &lt;- unique(seuratM$diseasecond)

table(seuratM$dataset)</code></pre>
<pre><code>
      344261_07-7_20240219_Hu_nucseq_control_01           344261_08-8_20240219_Hu_nucseq_UIA_04 
                                            598                                             398 
          344261_09-9_20240219_Hu_nucseq_UIA_07       348751_2-7_20240326_Hu_nucseq_tArm_ctrl_1 
                                           1132                                            1081 
353921_25-25_20240529_Hu_nucseq_Cadaver_ctrl_01     353921_27-27_20240522_Hu_nucseq_Ocp_ctrl_03 
                                           2270                                             487 
    359861_19-11_20240730_Hu_nucseq_Cad39_Aneur     359861_20-12_20240730_Hu_nucseq_Cad23_Aneur 
                                           2116                                            1454 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12 
2191 1316 1176 1027  940  920  510  447  331  297  230  103   48 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
     
9536 </code></pre>
</div>
<div id="plot-cell-numbers" class="section level2">
<h2>plot cell numbers</h2>
<pre class="r"><code>cell_count &lt;- data.frame(table(seuratM$patient))
colnames(cell_count) &lt;- c(&quot;patient&quot;, &quot;Freq&quot;)
hsize &lt;- 1.5

ggplot(cell_count, aes(x = hsize, y = Freq, fill = patient)) +
  scale_fill_manual(values = colpat2) +
  geom_col(color = &quot;white&quot;) +
  coord_polar(theta = &quot;y&quot;) +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle(&quot;cell number&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/cellnumber%20patient%20circle-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cell_count &lt;- data.frame(table(seuratM$diseasecond))
colnames(cell_count) &lt;- c(&quot;diseasecond&quot;, &quot;Freq&quot;)
hsize &lt;- 1.5

ggplot(cell_count, aes(x = hsize, y = Freq, fill = diseasecond)) +
  scale_fill_manual(values = coldiseasecond) +
  geom_col(color = &quot;white&quot;) +
  coord_polar(theta = &quot;y&quot;) +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle(&quot;cell number&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/cellnumber%20diseasecond%20circle-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-umaps" class="section level2">
<h2>plot umaps</h2>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.25
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.8) </code></pre>
<p><img src="figure/HumanAneurysm.Rmd/umap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-umap-1">
Past versions of umap-1.png
</button>
</p>
<div id="fig-umap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/umap-1.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d6526d4ac2dc949b6fc2e4608d44487e8541529c/docs/figure/HumanAneurysm.Rmd/umap-1.png" target="_blank">d6526d4</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$patient
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.8, cols = colpat2) </code></pre>
<p><img src="figure/HumanAneurysm.Rmd/umap-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-umap-2">
Past versions of umap-2.png
</button>
</p>
<div id="fig-umap-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/umap-2.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d6526d4ac2dc949b6fc2e4608d44487e8541529c/docs/figure/HumanAneurysm.Rmd/umap-2.png" target="_blank">d6526d4</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$diseasecond
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.8, cols = coldiseasecond) </code></pre>
<p><img src="figure/HumanAneurysm.Rmd/umap-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-marker-genes" class="section level2">
<h2>calculate marker genes</h2>
<pre class="r"><code>##cluster marker
Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.25
markerGenes &lt;- FindAllMarkers(seuratM, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#save table
write.table(markerGenes, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenes&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
<div id="integrate-over-batches" class="section level40">
<p class="heading">integrate over batches</p>
</div>
</div>
<div id="integrate-over-batches-1" class="section level2">
<h2>integrate over batches</h2>
<pre class="r"><code>seuratM.list &lt;- SplitObject(object = seuratM, split.by = &quot;batch&quot;)
for (i in 1:length(x = seuratM.list)) {
    seuratM.list[[i]] &lt;- NormalizeData(object = seuratM.list[[i]],
                                      verbose = FALSE)
    seuratM.list[[i]] &lt;- FindVariableFeatures(object = seuratM.list[[i]], 
        selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)
}

seuratM.anchors &lt;- FindIntegrationAnchors(object.list = seuratM.list, dims = 1:10,
                                         k.filter = 200)
seuratM.int &lt;- IntegrateData(anchorset = seuratM.anchors, dims = 1:10)
DefaultAssay(object = seuratM.int) &lt;- &quot;integrated&quot;

# rerun seurat
seuratM.int &lt;- ScaleData(object = seuratM.int, verbose = FALSE,
                        features = rownames(seuratM.int))
seuratM.int &lt;- RunPCA(object = seuratM.int, npcs = 20, verbose = FALSE)
#seuratM.int &lt;- RunTSNE(object = seuratM.int, reduction = &quot;pca&quot;, dims = 1:10)
seuratM.int &lt;- RunUMAP(object = seuratM.int, reduction = &quot;pca&quot;, dims = 1:10)

seuratM.int &lt;- FindNeighbors(object = seuratM.int, reduction = &quot;pca&quot;, dims = 1:10)
res &lt;- c(0.25,0.6,0.4,0.8)
for(i in 1:length(res)){
  seuratM.int &lt;- FindClusters(object = seuratM.int, resolution = res[i],
                             random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 316531

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9000
Number of communities: 8
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 316531

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8552
Number of communities: 13
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 316531

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8772
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 316531

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8373
Number of communities: 16
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>DefaultAssay(object = seuratM.int) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="visualize-data-integrated-over-batches" class="section level2">
<h2>visualize data integrated over batches</h2>
<pre class="r"><code>table(seuratM.int$integrated_snn_res.0.25)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7 
3296 1714 1437  933  890  805  273  188 </code></pre>
<pre class="r"><code>col &lt;- c(&quot;#BE3144&quot;,&quot;#f4a582&quot;,&quot;#B45B5C&quot;,&quot;#dfc27d&quot;,&quot;#779d8d&quot;,&quot;#3288BD&quot;,&quot;#355C7D&quot;, &quot;#ABDDA4&quot;)
names(col) &lt;- c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;)

Idents(seuratM.int) &lt;- seuratM.int$integrated_snn_res.0.25
DimPlot(seuratM.int, reduction=&quot;umap&quot;, pt.size = 1, cols = col)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20batches-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-batches-1">
Past versions of "visualize int cluster over batches-1.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-batches-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over batches-1.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM.int) &lt;- seuratM.int$batch
DimPlot(seuratM.int, reduction=&quot;umap&quot;, cols = colbatch, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20batches-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-batches-2">
Past versions of "visualize int cluster over batches-2.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-batches-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over batches-2.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM.int) &lt;- seuratM.int$patient
DimPlot(seuratM.int, reduction=&quot;umap&quot;, cols = colpat2, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20batches-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-batches-3">
Past versions of "visualize int cluster over batches-3.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-batches-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over batches-3.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM.int) &lt;- seuratM.int$diseasecond
DimPlot(seuratM.int, reduction=&quot;umap&quot;, cols = coldiseasecond, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20batches-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratM.int, reduction=&quot;umap&quot;, cols = coldiseasecond, pt.size = 1, split.by = &quot;diseasecond&quot;)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20batches-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colpat &lt;- c(&quot;#9E0142&quot;, &quot;#D53E4F&quot;, &quot;#F46D43&quot;, &quot;#FDAE61&quot;, &quot;#FEE08B&quot;, &quot;#E6F598&quot;, &quot;#ABDDA4&quot;, &quot;#66C2A5&quot;, &quot;#3288BD&quot;, &quot;#5E4FA2&quot; ,&quot;#BEAED4&quot; , &quot;#FDC086&quot;, &quot;#727077&quot;,&quot;#B45B5C&quot;,&quot;#202547&quot;)</code></pre>
</div>
<div id="calculate-marker-genes-integrated-over-batches"
class="section level2">
<h2>calculate marker genes integrated over batches</h2>
<pre class="r"><code>##cluster marker
Idents(seuratM.int) &lt;- seuratM.int$integrated_snn_res.0.25
markerGenes &lt;- FindAllMarkers(seuratM.int, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#save table
write.table(markerGenes, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenesintbatch&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="plot-features-intbatch" class="section level2">
<h2>plot features intbatch</h2>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000133392.MYH11&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000107796.ACTA2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000172403.SYNPO2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000161638.ITGA5&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000187498.COL4A1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000175084.DES&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000038427.VCAN&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000133110.POSTN&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000123358.NR4A1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000197956.S100A6&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000149591.TAGLN&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000026025.VIM&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000214548.MEG3&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000196569.LAMA2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000000971.CFH&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000081237.PTPRC&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000177575.CD163&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000010327.STAB1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000168685.IL7R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000116824.CD2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000261371.PECAM1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000172889.EGFL7&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000110799.VWF&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000107147.KCNT1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000168824.NSG1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM.int, features = &quot;ENSG00000164188.RANBP3L&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/feature%20plot%20intbatch-26.png" width="672" style="display: block; margin: auto;" />
## abundance plots int</p>
<pre class="r"><code>###patient
datList &lt;- NULL
for(con in unique(seuratM.int$patient)){
  seuratSub &lt;- subset(seuratM.int, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$integrated_snn_res.0.25)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 32435   598
[1] 32435   398
[1] 32435  1132
[1] 32435  1081
[1] 32435  2270
[1] 32435   487
[1] 32435  2116
[1] 32435  1454</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

ordpat &lt;- c(&quot;Ocpctrl03&quot;, &quot;ctrl01&quot;, &quot;tArmctrl1&quot;, &quot;Cadctrl01&quot;, &quot;Cad23&quot;, &quot;Cad39&quot;,&quot;UIA07&quot;, &quot;UIA04&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = col, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=ordpat)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/abundance%20plot%20intbatch-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###diseasecond
datList &lt;- NULL
for(con in unique(seuratM.int$diseasecond)){
  seuratSub &lt;- subset(seuratM.int, diseasecond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$integrated_snn_res.0.25)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseasecond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 32435  2166
[1] 32435  3646
[1] 32435  3724</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

orddiseasecond &lt;- c(&quot;ECctrl&quot;, &quot;ICctrl&quot;, &quot;aneurysm&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;diseasecond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = col, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=orddiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/abundance%20plot%20intbatch-2.png" width="960" style="display: block; margin: auto;" />
## subset fb</p>
<pre class="r"><code>seuratM.intFb &lt;- subset(seuratM.int,  integrated_snn_res.0.25 %in% c(&quot;0&quot;,&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;))
table(seuratM.intFb$integrated_snn_res.0.25)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7 
3296 1714 1437  933  890    0    0    0 </code></pre>
</div>
<div id="violin-plots" class="section level2">
<h2>violin plots</h2>
<pre class="r"><code>seuratM.intFb$diseasecond &lt;- factor(seuratM.intFb$diseasecond, levels = c(&quot;ECctrl&quot;,&quot;ICctrl&quot;,&quot;aneurysm&quot;))
Idents(seuratM.intFb) &lt;- seuratM.intFb$diseasecond
VlnPlot(object=seuratM.intFb, features = &quot;ENSG00000107796.ACTA2&quot;, pt.size = 0, cols = coldiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/violin%20plots%20fb-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratM.intFb, features = &quot;ENSG00000133392.MYH11&quot;, pt.size = 0, cols = coldiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/violin%20plots%20fb-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratM.intFb, features = &quot;ENSG00000187498.COL4A1&quot;, pt.size = 0, cols = coldiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/violin%20plots%20fb-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratM.intFb, features = &quot;ENSG00000175084.DES&quot;, pt.size = 0, cols = coldiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/violin%20plots%20fb-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="end-integrate-over-batches" class="section level40">
<p class="heading">end integrate over batches</p>
</div>
<div id="integrate-over-patients" class="section level40">
<p class="heading">integrate over patients</p>
</div>
</div>
<div id="integrate-over-patients-1" class="section level2">
<h2>integrate over patients</h2>
<pre class="r"><code>seuratM.list &lt;- SplitObject(object = seuratM, split.by = &quot;patient&quot;)
for (i in 1:length(x = seuratM.list)) {
    seuratM.list[[i]] &lt;- NormalizeData(object = seuratM.list[[i]],
                                      verbose = FALSE)
    seuratM.list[[i]] &lt;- FindVariableFeatures(object = seuratM.list[[i]], 
        selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)
}

seuratM.anchors &lt;- FindIntegrationAnchors(object.list = seuratM.list, dims = 1:10,
                                         k.filter = 200)
seuratM.intpat &lt;- IntegrateData(anchorset = seuratM.anchors, dims = 1:10)
DefaultAssay(object = seuratM.intpat) &lt;- &quot;integrated&quot;

# rerun seurat
seuratM.intpat &lt;- ScaleData(object = seuratM.intpat, verbose = FALSE,
                        features = rownames(seuratM.intpat))
seuratM.intpat &lt;- RunPCA(object = seuratM.intpat, npcs = 20, verbose = FALSE)
#seuratM.int &lt;- RunTSNE(object = seuratM.int, reduction = &quot;pca&quot;, dims = 1:10)
seuratM.intpat &lt;- RunUMAP(object = seuratM.intpat, reduction = &quot;pca&quot;, dims = 1:10)

seuratM.intpat &lt;- FindNeighbors(object = seuratM.intpat, reduction = &quot;pca&quot;, dims = 1:10)
res &lt;- c(0.25,0.6,0.4,0.8)
for(i in 1:length(res)){
  seuratM.intpat &lt;- FindClusters(object = seuratM.intpat, resolution = res[i],
                             random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 275973

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8577
Number of communities: 6
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 275973

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7813
Number of communities: 9
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 275973

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8194
Number of communities: 7
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9536
Number of edges: 275973

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7570
Number of communities: 10
Elapsed time: 1 seconds</code></pre>
<pre class="r"><code>DefaultAssay(object = seuratM.intpat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="visualize-data-integrated-over-patients"
class="section level2">
<h2>visualize data integrated over patients</h2>
<pre class="r"><code>#set color vector
col &lt;- c(&quot;#BE3144&quot;,&quot;#f4a582&quot;,&quot;#B45B5C&quot;,&quot;#dfc27d&quot;,&quot;#779d8d&quot;,&quot;#3288BD&quot;)
names(col) &lt;- c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;)

Idents(seuratM.intpat) &lt;- seuratM.intpat$integrated_snn_res.0.25
DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = col, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-1">
Past versions of "visualize int cluster over patients-1.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-1.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = col, pt.size = 1) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-2">
Past versions of "visualize int cluster over patients-2.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-2.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM.intpat) &lt;- seuratM.intpat$batch
DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = colbatch, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-3">
Past versions of "visualize int cluster over patients-3.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-3.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = colbatch, pt.size = 1) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-4">
Past versions of "visualize int cluster over patients-4.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-4.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratM.intpat) &lt;- seuratM.intpat$patient
DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = colpat2, pt.size = 1)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-5">
Past versions of "visualize int cluster over patients-5.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-5.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratM.intpat, reduction=&quot;umap&quot;, cols = colpat2, pt.size = 1) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/visualize%20int%20cluster%20over%20patients-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-visualize-int-cluster-over-patients-6">
Past versions of "visualize int cluster over patients-6.png"
</button>
</p>
<div id="fig-no-spaces-visualize-int-cluster-over-patients-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/visualize int cluster over patients-6.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="calculate-marker-genes-integrated-over-patients"
class="section level2">
<h2>calculate marker genes integrated over patients</h2>
<pre class="r"><code>##cluster marker
Idents(seuratM.intpat) &lt;- seuratM.intpat$integrated_snn_res.0.25
markerGenes &lt;- FindAllMarkers(seuratM, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#save table
write.table(markerGenes, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/HumanAneurysm/analysis/markerGenesintpat&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="abundance-plots" class="section level2">
<h2>abundance plots</h2>
<pre class="r"><code>###patient
datList &lt;- NULL
for(con in unique(seuratM.intpat$patient)){
  seuratSub &lt;- subset(seuratM.intpat, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$integrated_snn_res.0.25)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 32435   598
[1] 32435   398
[1] 32435  1132
[1] 32435  1081
[1] 32435  2270
[1] 32435   487
[1] 32435  2116
[1] 32435  1454</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

ordpat &lt;- c(&quot;Ocpctrl03&quot;, &quot;ctrl01&quot;, &quot;tArmctrl1&quot;, &quot;Cadctrl01&quot;, &quot;Cad23&quot;, &quot;Cad39&quot;,&quot;UIA07&quot;, &quot;UIA04&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = col, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=ordpat)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/abundance%20plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-abundance-plot-1">
Past versions of "abundance plot-1.png"
</button>
</p>
<div id="fig-no-spaces-abundance-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/HumanAneurysm/blob/d91538dcf89555aae858fc6bfcc58c965538e570/docs/figure/HumanAneurysm.Rmd/abundance plot-1.png" target="_blank">d91538d</a>
</td>
<td>
angeldemartin
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###diseasecond
datList &lt;- NULL
for(con in unique(seuratM.intpat$diseasecond)){
  seuratSub &lt;- subset(seuratM.intpat, diseasecond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$integrated_snn_res.0.25)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseasecond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 32435  2166
[1] 32435  3646
[1] 32435  3724</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

orddiseasecond &lt;- c(&quot;ECctrl&quot;, &quot;ICctrl&quot;, &quot;aneurysm&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;diseasecond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = col, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=orddiseasecond)</code></pre>
<p><img src="figure/HumanAneurysm.Rmd/abundance%20plot-2.png" width="960" style="display: block; margin: auto;" />
######################################## end integrate over patients
########################################################</p>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri Aug 23 14:14:35 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.6.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
 [4] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
 [7] DOSE_3.30.1                 enrichplot_1.24.0           msigdbr_7.5.1              
[10] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.0     
[13] multtest_2.60.0             metap_1.11                  scater_1.32.0              
[16] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[19] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[22] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[25] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[28] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[31] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[34] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[37] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[40] matrixStats_1.3.0           pheatmap_1.0.12             ggpubr_0.6.0               
[43] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[46] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] zlibbioc_1.50.0           tidyselect_1.2.1          bit_4.0.5                
  [7] doParallel_1.0.17         clue_0.3-65               lattice_0.22-6           
 [10] rjson_0.2.21              blob_1.2.4                S4Arrays_1.4.1           
 [13] pbkrtest_0.5.3            parallel_4.4.0            png_0.1-8                
 [16] plotrix_3.8-4             cli_3.6.3                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] BiocNeighbors_1.22.0      uwot_0.2.2                shadowtext_0.1.3         
 [25] curl_5.2.1                mime_0.12                 evaluate_0.24.0          
 [28] tidytree_0.4.6            leiden_0.4.3.1            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.4             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] splines_4.4.0             ggraph_2.2.1              sctransform_0.4.1        
 [40] ggbeeswarm_0.7.2          DBI_1.2.3                 jquerylib_0.1.4          
 [43] smoother_1.3              withr_3.0.0               git2r_0.33.0             
 [46] corpcor_1.6.10            class_7.3-22              rprojroot_2.0.4          
 [49] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [61] DEoptimR_1.1-3            reticulate_1.38.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-1           irlba_2.3.5.1            
 [79] ggrastr_1.0.2             fastDummies_1.7.3         gridGraphics_0.5-1       
 [82] lazyeval_0.2.2            yaml_2.3.9                survival_3.7-0           
 [85] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [88] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [91] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [94] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
 [97] shape_1.4.6.1             limma_3.60.3              pkgconfig_2.0.3          
[100] TMB_1.9.14                mathjaxr_1.6-0            EnvStats_2.8.1           
[103] aplot_0.2.3               scatterplot3d_0.3-44      spatstat.sparse_3.1-0    
[106] ape_5.8                   xtable_1.8-4              car_3.1-2                
[109] highr_0.11                plyr_1.8.9                httr_1.4.7               
[112] rbibutils_2.2.16          tools_4.4.0               globals_0.16.3           
[115] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-165             
[118] lambda.r_1.2.4            HDO.db_0.99.1             lme4_1.1-35.5            
[121] digest_0.6.36             numDeriv_2016.8-1.1       Matrix_1.7-0             
[124] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[127] reshape2_1.4.4            yulab.utils_0.1.4         glue_1.7.0               
[130] cachem_1.1.0              polyclip_1.10-6           generics_0.1.3           
[133] Biostrings_2.72.1         mvtnorm_1.2-5             parallelly_1.37.1        
[136] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[139] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.7              
[142] pbapply_1.7-2             spam_2.10-0               gson_0.1.0               
[145] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[148] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[151] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[154] rmarkdown_2.27            scales_1.3.0              future_1.33.2            
[157] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-2      
[160] rstudioapi_0.16.0         cluster_2.1.6             whisker_0.4.1            
[163] mutoss_0.1-13             spatstat.utils_3.0-5      hms_1.1.3                
[166] fitdistrplus_1.1-11       munsell_0.5.1             cowplot_1.1.3            
[169] colorspace_2.1-0          rlang_1.1.4               DelayedMatrixStats_1.26.0
[172] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.1-1          
[175] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[178] mgcv_1.9-1                xfun_0.45                 e1071_1.7-14             
[181] TH.data_1.1-2             iterators_1.0.14          abind_1.4-5              
[184] GOSemSim_2.30.0           treeio_1.28.0             futile.options_1.0.1     
[187] bitops_1.0-7              Rdpack_2.6                promises_1.3.0           
[190] scatterpie_0.2.3          RSQLite_2.3.7             qvalue_2.36.0            
[193] sandwich_3.1-0            fgsea_1.30.0              DelayedArray_0.30.1      
[196] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.0           
[199] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[202] listenv_0.9.1             Rcpp_1.0.12               edgeR_4.2.0              
[205] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[208] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[211] babelgene_22.9            spatstat.random_3.2-3     R6_2.5.1                 
[214] fastmap_1.2.0             multcomp_1.4-25           fastmatch_1.1-4          
[217] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[220] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[223] vcd_1.4-12                nnet_7.3-19               gtable_0.3.5             
[226] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[229] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.0.5              
[232] spatstat.explore_3.2-7    lifecycle_1.0.4           blme_1.0-5               
[235] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[238] robustbase_0.99-3         spatstat.geom_3.2-9       sn_2.1.1                 
[241] ggfun_0.1.5               future.apply_1.11.2       bslib_0.7.0              
[244] pillar_1.9.0              gplots_3.1.3.1            pcaMethods_1.96.0        
[247] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.6.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
 [4] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
 [7] DOSE_3.30.1                 enrichplot_1.24.0           msigdbr_7.5.1              
[10] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.0     
[13] multtest_2.60.0             metap_1.11                  scater_1.32.0              
[16] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[19] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[22] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[25] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[28] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[31] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[34] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[37] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[40] matrixStats_1.3.0           pheatmap_1.0.12             ggpubr_0.6.0               
[43] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[46] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] zlibbioc_1.50.0           tidyselect_1.2.1          bit_4.0.5                
  [7] doParallel_1.0.17         clue_0.3-65               lattice_0.22-6           
 [10] rjson_0.2.21              blob_1.2.4                S4Arrays_1.4.1           
 [13] pbkrtest_0.5.3            parallel_4.4.0            png_0.1-8                
 [16] plotrix_3.8-4             cli_3.6.3                 ggplotify_0.1.2          
 [19] goftest_1.2-3             VIM_6.2.2                 variancePartition_1.34.0 
 [22] BiocNeighbors_1.22.0      uwot_0.2.2                shadowtext_0.1.3         
 [25] curl_5.2.1                mime_0.12                 evaluate_0.24.0          
 [28] tidytree_0.4.6            leiden_0.4.3.1            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.4             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] splines_4.4.0             ggraph_2.2.1              sctransform_0.4.1        
 [40] ggbeeswarm_0.7.2          DBI_1.2.3                 jquerylib_0.1.4          
 [43] smoother_1.3              withr_3.0.0               git2r_0.33.0             
 [46] corpcor_1.6.10            class_7.3-22              rprojroot_2.0.4          
 [49] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [52] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [55] ggrepel_0.9.5             labeling_0.4.3            fANCOVA_0.6-1            
 [58] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [61] DEoptimR_1.1-3            reticulate_1.38.0         hexbin_1.28.3            
 [64] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [67] ggplot.multistats_1.0.0   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [70] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [73] patchwork_1.2.0           caTools_1.18.2            data.table_1.15.4        
 [76] ggtree_3.12.0             RSpectra_0.16-1           irlba_2.3.5.1            
 [79] ggrastr_1.0.2             fastDummies_1.7.3         gridGraphics_0.5-1       
 [82] lazyeval_0.2.2            yaml_2.3.9                survival_3.7-0           
 [85] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [88] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [91] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [94] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
 [97] shape_1.4.6.1             limma_3.60.3              pkgconfig_2.0.3          
[100] TMB_1.9.14                mathjaxr_1.6-0            EnvStats_2.8.1           
[103] aplot_0.2.3               scatterplot3d_0.3-44      spatstat.sparse_3.1-0    
[106] ape_5.8                   xtable_1.8-4              car_3.1-2                
[109] highr_0.11                plyr_1.8.9                httr_1.4.7               
[112] rbibutils_2.2.16          tools_4.4.0               globals_0.16.3           
[115] beeswarm_0.4.0            broom_1.0.6               nlme_3.1-165             
[118] lambda.r_1.2.4            HDO.db_0.99.1             lme4_1.1-35.5            
[121] digest_0.6.36             numDeriv_2016.8-1.1       Matrix_1.7-0             
[124] farver_2.1.2              tzdb_0.4.0                remaCor_0.0.18           
[127] reshape2_1.4.4            yulab.utils_0.1.4         glue_1.7.0               
[130] cachem_1.1.0              polyclip_1.10-6           generics_0.1.3           
[133] Biostrings_2.72.1         mvtnorm_1.2-5             parallelly_1.37.1        
[136] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[139] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.7              
[142] pbapply_1.7-2             spam_2.10-0               gson_0.1.0               
[145] utf8_1.2.4                graphlayouts_1.1.1        gtools_3.9.5             
[148] ggsignif_0.6.4            RcppEigen_0.3.4.0.0       shiny_1.8.1.1            
[151] GenomeInfoDbData_1.2.12   glmmTMB_1.1.9             memoise_2.0.1            
[154] rmarkdown_2.27            scales_1.3.0              future_1.33.2            
[157] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-2      
[160] rstudioapi_0.16.0         cluster_2.1.6             whisker_0.4.1            
[163] mutoss_0.1-13             spatstat.utils_3.0-5      hms_1.1.3                
[166] fitdistrplus_1.1-11       munsell_0.5.1             cowplot_1.1.3            
[169] colorspace_2.1-0          rlang_1.1.4               DelayedMatrixStats_1.26.0
[172] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.1-1          
[175] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[178] mgcv_1.9-1                xfun_0.45                 e1071_1.7-14             
[181] TH.data_1.1-2             iterators_1.0.14          abind_1.4-5              
[184] GOSemSim_2.30.0           treeio_1.28.0             futile.options_1.0.1     
[187] bitops_1.0-7              Rdpack_2.6                promises_1.3.0           
[190] scatterpie_0.2.3          RSQLite_2.3.7             qvalue_2.36.0            
[193] sandwich_3.1-0            fgsea_1.30.0              DelayedArray_0.30.1      
[196] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.0           
[199] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[202] listenv_0.9.1             Rcpp_1.0.12               edgeR_4.2.0              
[205] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[208] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[211] babelgene_22.9            spatstat.random_3.2-3     R6_2.5.1                 
[214] fastmap_1.2.0             multcomp_1.4-25           fastmatch_1.1-4          
[217] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[220] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[223] vcd_1.4-12                nnet_7.3-19               gtable_0.3.5             
[226] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[229] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.0.5              
[232] spatstat.explore_3.2-7    lifecycle_1.0.4           blme_1.0-5               
[235] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[238] robustbase_0.99-3         spatstat.geom_3.2-9       sn_2.1.1                 
[241] ggfun_0.1.5               future.apply_1.11.2       bslib_0.7.0              
[244] pillar_1.9.0              gplots_3.1.3.1            pcaMethods_1.96.0        
[247] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
